File: push-openapi-nodejs-demo/Custom_Signature/request.js
var sign = require('./sign.js');
var popEscape = require('./popEscape.js');

module.exports = {
    request : function(host,params,secret){
	var keys = Object.keys(params);
        var param = '';
	for (key in keys) {
    	    param = param + popEscape.popEscape(keys[key]) + "=" + popEscape.popEscape(params[keys[key]]) + "&";
	}
        request = host + "/?" + param + "Signature=" + popEscape.popEscape(sign.sign("GET",params,secret));
	return request;
    }
};

File: OpenApi1.0/push-openapi-net-demo/Custom_Signature/Request.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace AliCloudPush
{
    class Request
    {
        public static string request(string host, SortedDictionary<string, object> parameters,string secret)
        {
            StringBuilder queryString = new StringBuilder();
            foreach (var p in parameters)
            {
                queryString
                .Append(PopEncode.Encode(p.Key)).Append("=")
                .Append(PopEncode.Encode(p.Value.ToString()))
                .Append("&");
            }
            string request = host + "/?" + queryString + "Signature=" + Sign.sign("GET",parameters,secret+"&");
            return request;
        }
    }
}
